Tests/EmbeddedScripts.CSharp.MonoEvaluator.Tests/MonoCSharpRunnerMultithreadingTests.cs

```diff
@@ -0,0 +1,203 @@
+﻿using System;
+using System.Threading.Tasks;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace EmbeddedScripts.CSharp.MonoEvaluator.Tests
+{
+    public class MonoCSharpRunnerMultithreadingTests
+    {
+        private readonly ITestOutputHelper _testOutputHelper;
+
+        public MonoCSharpRunnerMultithreadingTests(ITestOutputHelper testOutputHelper)
+        {
+            _testOutputHelper = testOutputHelper;
+        }
+
+        [Fact]
+        public async Task EvaluateAsync_RunParallelTasksWithOneRunner_Success()
+        {
+            var code = @"
+class A
+{
+  public static long fib(int n)
+  {
+    if (n == 0 || n == 1)
+      return n;
+    return fib(n - 1) + fib(n - 2);
+  }
+
+  public static long start(int n, int id)
+  {
+    log($""start fib {id}"");
+    long res = fib(n);
+    log($""end fib {id}"");
+    return res;
+  }
+}
+";
+
+            var runner = new MonoCSharpRunner();
+            runner.Register<Action<string>>(s =>
+            {
+                _testOutputHelper.WriteLine($"At {DateTime.Now}");
+                _testOutputHelper.WriteLine(s);
+            }, "log");
+            
+            await runner.RunAsync(code);
+
+            var firstTask = Task.Run(async () =>
+            {
+                var t = runner.EvaluateAsync<long>("A.start(40, 1)");
+                await t;
+            });
+
+            var secondTask = Task.Run(async () =>
+            {
+                var t = runner.EvaluateAsync<long>("A.start(40, 2)");
+                await t;
+            });
+
+            await Task.WhenAll(firstTask, secondTask);
+        }
+
+        [Fact]
+        public async Task Evaluate_WhileEvaluateIsRunningInSeparateTask_Success() 
+        {
+            var code = @"
+class A
```

Comment:
Does 3 tests contain the same fibonacci script in this class? Maybe extract code as string constant